src/main/java/mate/jdbc/dao/impl/CarDaoImpl.java

```diff
@@ -0,0 +1,206 @@
+        String query = "DELETE FROM cars_drivers WHERE car_id = ?";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(query)) {
+            statement.setLong(1, carId);
+            statement.executeUpdate();
+        } catch (SQLException e) {
+            throw new DataProcessingException(
+                    "Can't delete cars_drivers relationship by id " + carId, e);
+        }
+    }
+
+    private void addCarDriversRelationship(Car car) {
+        String query = "INSERT INTO cars_drivers (car_id, driver_id) VALUES (?, ?)";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(query)) {
+            statement.setLong(1, car.getId());
+            List<Driver> driversList = car.getDriversList();
+            for (Driver driver : driversList) {
+                statement.setLong(2, driver.getId());
+                statement.executeUpdate();
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException(
+                    "Can't insert new cars_drivers relationship by car " + car, e);
+        }
+    }
+
+    @Override
+    public boolean delete(Long id) {
+        String query = "DELETE FROM cars WHERE id = ?";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(query)) {
+            statement.setLong(1, id);
+            deleteCarDriversRelationship(id);
+            return statement.executeUpdate() > 0;
+        } catch (SQLException e) {
+            throw new DataProcessingException(
+                    "Can't delete the car by id " + id, e);
+        }
+    }
+
+    @Override
+    public List<Car> getAllByDriver(Long driverId) {
+        String query = "SELECT id, manufacturer_id, model "
+                + "FROM cars "
+                + "JOIN cars_drivers cd "
+                + "ON cars.id = cd.car_id "
+                + "WHERE cd.driver_id = ?";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(query)) {
+            statement.setLong(1, driverId);
+            ResultSet resultSet = statement.executeQuery();
+
+            ManufacturerDao manufacturerDao = new ManufacturerDaoImpl();
+            List<Car> carsList = new ArrayList<>();
+            while (resultSet.next()) {
+                Long id = resultSet.getObject("id", Long.class);
+                Manufacturer manufacturer = manufacturerDao.get(
+                        resultSet.getObject("manufacturer_id", Long.class))
+                        .orElse(null);
+                String model = resultSet.getString("model");
+                List<Driver> driversList = getDriversByCarId(id);
+                Car car = new Car(id, model, manufacturer, driversList);
```

Comment:
what about Car car = getCar(resultSet) method? we can use it here, if I'm not mistaken 
```suggestion
                Car car = getCar(resultSet);
                carsList.add(car);
```